src/Opdex.Platform.Application.Abstractions/Queries/Tokens/Snapshots/RetrieveTokenSnapshotsWithFilterQuery.cs

```diff
@@ -1,64 +1,36 @@
 using System;
 using System.Collections.Generic;
 using MediatR;
-using Opdex.Platform.Common.Enums;
-using Opdex.Platform.Common.Extensions;
 using Opdex.Platform.Domain.Models.Tokens;
+using Opdex.Platform.Infrastructure.Abstractions.Data.Queries;
 
 namespace Opdex.Platform.Application.Abstractions.Queries.Tokens.Snapshots
 {
     /// <summary>
-    /// Retrieve token snapshots by a token and its market, their type and date range else an empty list.
+    /// Retrieves token snapshots for a given token.
     /// </summary>
     public class RetrieveTokenSnapshotsWithFilterQuery : IRequest<IEnumerable<TokenSnapshot>>
     {
         /// <summary>
-        /// Create the retrieve token snapshots with filter query.
+        /// Creates a request to retrieve snapshot data for a given token.
         /// </summary>
         /// <param name="tokenId">The token id of snapshots to find.</param>
-        /// <param name="marketId">The market id of snapshots to find.</param>
-        /// <param name="startDate">The start date, earliest snapshot to find.</param>
-        /// <param name="endDate">The end date, latest snapshot to find.</param>
-        /// <param name="snapshotType">The type of snapshots requested, CRS is minute/hourly/daily, SRC is hourly/daily.</param>
-        public RetrieveTokenSnapshotsWithFilterQuery(ulong tokenId, ulong marketId, DateTime startDate, DateTime endDate, SnapshotType snapshotType)
+        /// <param name="marketId">The market id of snapshots to find, or 0 for global snapshots.</param>
+        /// <param name="cursor">The snapshot cursor filter.</param>
+        public RetrieveTokenSnapshotsWithFilterQuery(ulong tokenId, ulong marketId, SnapshotCursor cursor)
         {
-            if (tokenId < 1)
-            {
-                throw new ArgumentOutOfRangeException(nameof(tokenId));
-            }
-
+            if (tokenId < 1) throw new ArgumentOutOfRangeException(nameof(tokenId));
             // 0 is valid
-            if (marketId < 0)
-            {
-                throw new ArgumentOutOfRangeException(nameof(marketId), "Market id must be greater or equal to zero.");
-            }
-
-            if (startDate.Equals(default))
-            {
-                throw new ArgumentOutOfRangeException(nameof(startDate));
-            }
-
-            if (endDate.Equals(default) || endDate < startDate)
-            {
-                throw new ArgumentOutOfRangeException(nameof(endDate));
-            }
-
-            if (!snapshotType.IsValid())
-            {
-                throw new ArgumentOutOfRangeException(nameof(snapshotType));
-            }
+            if (marketId < 0) throw new ArgumentOutOfRangeException(nameof(marketId), "Market id must be greater or equal to zero.");
```

Comment:
This can probably be removed now that `marketId` is `ulong` I think the original check for `< 0` was when we were using `long` for Ids.

Reply:
Cool, removed these checks